Src/HALALMock/Services/Communication/SPI/SPI.cpp

```diff
@@ -31,30 +31,77 @@ uint8_t SPI::inscribe(SPI::Peripheral& spi) {
 
     SPI::Instance* spi_instance = SPI::available_spi[spi];
 
-    // this three lines don t do nothing yet, as the alternative_function on a
-    // pin is hardcoded on the hal_msp.c
-    spi_instance->SCK->alternative_function = AF5;
-    spi_instance->MOSI->alternative_function = AF5;
-    spi_instance->MISO->alternative_function = AF5;
-
-    spi_instance->SPIOrderID =
-        (uint16_t*)MPUManager::allocate_non_cached_memory(32);
-    spi_instance->available_end =
-        (uint16_t*)MPUManager::allocate_non_cached_memory(32);
-    spi_instance->rx_buffer = (uint8_t*)MPUManager::allocate_non_cached_memory(
-        SPI_MAXIMUM_PACKET_SIZE_BYTES);
-    spi_instance->tx_buffer = (uint8_t*)MPUManager::allocate_non_cached_memory(
-        SPI_MAXIMUM_PACKET_SIZE_BYTES);
-
-    Pin::inscribe(*spi_instance->SCK, ALTERNATIVE);
-    Pin::inscribe(*spi_instance->MOSI, ALTERNATIVE);
-    Pin::inscribe(*spi_instance->MISO, ALTERNATIVE);
+    EmulatedPin& SCK_pin = SharedMemory::get_pin(*spi_instance->SCK);
+    EmulatedPin& MOSI_pin = SharedMemory::get_pin(*spi_instance->MOSI);
+    EmulatedPin& MISO_pin = SharedMemory::get_pin(*spi_instance->MISO);
 
-    if (spi_instance->mode == SPI_MODE_MASTER) {
-        Pin::inscribe(*spi_instance->SS, OUTPUT);
+    if (SCK_pin.type != PinType::NOT_USED ||
+        MOSI_pin.type != PinType::NOT_USED ||
+        MISO_pin.type != PinType::NOT_USED) {
+        ErrorHandler("The SPI pins are already used");
+        return 0;
     }
 
+    SCK_pin.type = PinType::SPI;
+    MOSI_pin.type = PinType::SPI;
+    MISO_pin.type = PinType::SPI;
+
     uint8_t id = SPI::id_counter++;
+
+    // Create socket for this SPI peripheral
+    int spi_socket = socket(AF_INET, SOCK_STREAM, 0);
+
+    // Bind this socket to the local port 200x
+    sockaddr_in local_address;
+    local_address.sin_family = AF_INET;
+    local_address.sin_port = htons(SPI_PORT_BASE + id_counter);
+    local_address.sin_addr.s_addr = INADDR_ANY;
```

Comment:
address should be fixed, otherwise how are clients going to connect?

Reply:
But it is assumed that clients will be in the same host, isn't it?

Reply:
I looked into the documentation for INADDR_ANY, and yes that should work further info, [ understanding-inaddr-any-for-socket-programming](https://stackoverflow.com/questions/16508685/understanding-inaddr-any-for-socket-programming)

They will be in the same host yes, but maybe they have differents IPs, but INADDR_ANY bounds to **ALL** addresses so it works fine